doc/guides/west/build-flash-debug.rst

```diff
@@ -151,37 +161,44 @@ To let west decide for you if a pristine build is needed, use ``-p auto``::
    permanent.
 
 .. _west-building-generator:
+.. _west-building-cmake-args:
+
+Additional CMake Arguments
+--------------------------
 
-To add additional arguments to the CMake invocation performed by ``west
+To pass additional arguments to the CMake invocation performed by ``west
 build``, pass them after a ``--`` at the end of the command line.
 
+.. important::
+
+   Passing additional CMake arguments like this forces ``west build`` to re-run
+   CMake, even if a build system has already been generated.
+
+   After using ``--`` once to generate the build directory, use ``west build -d
+   <build-dir>`` on subsequent runs to do incremental builds.
+
 For example, to use the Unix Makefiles CMake generator instead of Ninja (which
 ``west build`` uses by default), run::
 
   west build -b reel_board -- -G'Unix Makefiles'
 
-.. note::
-
-   Passing additional CMake arguments like this forces ``west build`` to re-run
-   CMake, even if a build system has already been generated.
-
-As another example, to use Unix Makefiles and enable the
-`CMAKE_VERBOSE_MAKEFILE`_ option::
+To use Unix Makefiles and set `CMAKE_VERBOSE_MAKEFILE`_ to ``ON``::
 
   west build -b reel_board -- -G'Unix Makefiles' -DCMAKE_VERBOSE_MAKEFILE=ON
 
 Notice how the ``--`` only appears once, even though multiple CMake arguments
 are given. All command-line arguments to ``west build`` after a ``--`` are
 passed to CMake.
 
-As a final example, to merge the :file:`file.conf` Kconfig fragment into your
-build's :file:`.config`::
+To use the file :file:`enable-modem.overlay` as a :ref:`device tree overlay
+<device-tree>` by setting :ref:`DTC_OVERLAY_FILE <application_dt>`::
```

Comment:
s/by setting/set/

Reply:
I think the grammar is correct as is (to accomplish X by setting Y to Z), but it must be unclear. How about this?

```
To set :ref:`DTC_OVERLAY_FILE <application_dt>` to :file:`enable-modem.overlay`,
using that file as a :ref:`device tree overlay <device-tree>`::
```